ref-exchange/src/action.rs

```diff
@@ -8,13 +8,13 @@ pub struct SwapAction {
     /// Pool which should be used for swapping.
     pub pool_id: u64,
     /// Token to swap from.
-    pub token_in: ValidAccountId,
+    pub token_in: AccountId,
```

Comment:
we don't need to use `ValidAccountId` here because we check if the tokens are already registered. Since we can't register not valid acocunt / token, then we don't need to additional check and allocation.